### PR TITLE
added command line param to load replay file, made orders specific to…

### PR DESCRIPTION
**Pull request by DaanMeijer**
… player, disabled object removal so object ids match up more

### CODE REVIEW

File: src/OpenSage.Game/Gui/DrawingContext2D.cs [hunk 1]
```diff
@@ -109,14 +109,7 @@ private void UpdateOpacity()
...
         {
-            var color = ColorRgbaF.White.WithA(_currentOpacity);
-
-            _spriteBatch.DrawImage(
-                texture,
-                sourceRect,
-                RectangleF.Transform(destinationRect.ToRectangleF(), _currentTransform),
-                color,
-                flipped);
+            DrawImage(texture, sourceRect, destinationRect.ToRectangleF(), flipped);
```
Comment (feliwir) [inline src/OpenSage.Game/Gui/DrawingContext2D.cs]: Why you removed the opacity here?

File: src/OpenSage.Game/Logic/Object/GameObjectCollection.cs [hunk 2]
```diff
@@ -56,7 +56,9 @@ public GameObject GetObjectById(int objectId)
         public void Remove(GameObject gameObject)
         {
             // TODO: We should probably be using an ID.
-            _items.Remove(gameObject);
+            var index = _items.IndexOf(gameObject);
```
Comment (feliwir) [inline src/OpenSage.Game/Logic/Object/GameObjectCollection.cs]: Not used

File: src/OpenSage.Game/Logic/Orders/OrderProcessor.cs [hunk 3]
```diff
@@ -15,12 +16,24 @@ public OrderProcessor(Game game)
...
+        private static NLog.Logger logger = NLog.LogManager.GetCurrentClassLogger();
+
         public void Process(IEnumerable<Order> orders)
         {
+            
             foreach (var order in orders)
             {
                 var player = _game.Scene3D.Players[(int) order.PlayerIndex];
 
+                if(order.OrderType == OrderType.SetCameraPosition)
```
Comment (feliwir) [inline src/OpenSage.Game/Logic/Orders/OrderProcessor.cs]: You could probably compose the message outside of the if/else statement and then just print the string inside

File: src/OpenSage.Game/Logic/Orders/OrderProcessor.cs [hunk 4]
```diff
@@ -61,30 +89,80 @@ public void Process(IEnumerable<Order> orders)
...
+                                .Select(x => (int) x.Value.ObjectId)
+                                .Select(_game.Scene3D.GameObjects.GetObjectById)
+                                .ToArray();
 
-                        var objectIds = order.Arguments.Skip(1)
-                            .Select(x => (int) x.Value.ObjectId)
-                            .Select(_game.Scene3D.GameObjects.GetObjectById)
-                            .ToArray();
+                            _game.Selection.SetSelectedObjects(player, objectIds);
+                        }catch(System.Exception e)
```
Comment (feliwir) [inline src/OpenSage.Game/Logic/Orders/OrderProcessor.cs]: i think we usually put the catch into a newline

File: src/OpenSage.Launcher/Program.cs [hunk 5]
```diff
@@ -108,17 +112,18 @@ public static void Run(Options opts)
...
+                if (opts.ReplayFile != null)
                 {
-                    var pSettings = new[]
+                    using (var fileSystem = new FileSystem(Path.Combine(game.UserDataFolder, "Replays")))
                     {
-                        new PlayerSetting("America", new ColorRgb(255, 0, 0)),
-                        new PlayerSetting("GLA", new ColorRgb(255, 255, 255)),
+                        game.LoadReplayFile(fileSystem.GetFile(opts.ReplayFile));
+                    }
+                }else if (opts.Map != null)
```
Comment (feliwir) [inline src/OpenSage.Launcher/Program.cs]: formatting issue with the else if

File: src/OpenSage.Game/Game.cs [hunk 6]
```diff
@@ -86,6 +87,100 @@ public sealed class Game : DisposableBase
...
+                        var diff = maxFactionIndex - minFactionIndex;
+                        factionIndex = minFactionIndex + (random.Next() % diff);
+                    }
+
+                    var faction = ContentManager.IniDataContext.PlayerTemplates[factionIndex];
+
+                    Console.WriteLine(ContentManager.IniDataContext.PlayerTemplates);
+                    var color = new ColorRgb(0, 0, 0);
+                  
+                    switch (slot.Color)
```
Comment (feliwir) [inline src/OpenSage.Game/Game.cs]: I don't think those are hardcoded. Maybe check in MutiplayerColors or some other ini entry

File: src/OpenSage.Game/Logic/Object/GameObjectCollection.cs (hunk 2 above)
Comment (DaanMeijer) [reply src/OpenSage.Game/Logic/Object/GameObjectCollection.cs]: Fixed that, we probably really shouldn't remove objects from the collection. They come in handy later on, for statistics purposes

File: src/OpenSage.Game/Logic/Orders/OrderProcessor.cs (hunk 4 above)
Comment (DaanMeijer) [reply src/OpenSage.Game/Logic/Orders/OrderProcessor.cs]: Fixed that

File: src/OpenSage.Launcher/Program.cs (hunk 5 above)
Comment (DaanMeijer) [reply src/OpenSage.Launcher/Program.cs]: Fixed that

File: src/OpenSage.Game/Game.cs (hunk 6 above)
Comment (DaanMeijer) [reply src/OpenSage.Game/Game.cs]: colors get loaded from ini files now :)

File: src/OpenSage.Game/Gui/DrawingContext2D.cs (hunk 1 above)
Comment (DaanMeijer) [reply src/OpenSage.Game/Gui/DrawingContext2D.cs]: Because the overloaded function it calls calculates the opacity already

**Comment by feliwir** [comment]
If this is contained in #310 can we close this?

**Pull request closed without merging**